Docs: align the documentation and the examples to the new changes (#1181)
metallb/metallb: website/content/release-notes/_index.md

weight: 8
---

## Version 0.12.0

New Features:

- Experimental FRR mode is now available. In this mode, the BGP stack is handled
  by a FRR container in place of the native BGP implementation. This offers additional capabilities such as
  IPv6 BGP announcement and BFD support. See the installation section on how
  to enable it.
  ([PR #832](https://github.com/metallb/metallb/pull/832), [PR #935](https://github.com/metallb/metallb/pull/935), [PR #958](https://github.com/metallb/metallb/pull/958), [PR #1014](https://github.com/metallb/metallb/pull/1014) and others)

- Dual stack services are now supported. L2 works out of the box, BGP requires
  the FRR mode because of missing IPv6 support in the native implementation.
  ([PR #1065](https://github.com/metallb/metallb/pull/1065))

- In FRR mode, it is possible to have a BGP session paired with a BFD session
  for quicker path failure detection.
  ([PR #927](https://github.com/metallb/metallb/pull/927))
  ([PR #967](https://github.com/metallb/metallb/pull/967))

> - A new manifest (`metallb-frr.yaml`) is available to deploy metallb in FRR mode

Reviewer: Missed this suggestion?

```suggestion
- A new manifest (`manifests/metallb-frr.yaml`) is available to deploy MetalLB in FRR mode
```

Author: Done

weight: 8
---

## Version 0.12.0

New Features:

> - FRR Mode is now supported. An experimental launch mode where the bgp stack is handled
>   by a FRR container in place of the native bgp implementation. This enables
>   IPv6 BGP announcement and BFD Support. See the installation section on how
>   to enable it.

Reviewer: ```suggestion
- Experimental FRR mode is now available. In this mode, the BGP stack is handled
  by a FRR container in place of the native BGP implementation. This offers additional capabilities such as
  IPv6 BGP announcement and BFD support. See the installation section on how
  to enable it.
```

  ([PR #832](https://github.com/metallb/metallb/pull/832), [PR #935](https://github.com/metallb/metallb/pull/935), [PR #958](https://github.com/metallb/metallb/pull/958), [PR #1014](https://github.com/metallb/metallb/pull/1014) and others)

- Dual stack services are now supported. L2 works out of the box, BGP requires

>   the FRR mode because of missing ipv6 support in the native implementation.

Reviewer: ```suggestion
  the FRR mode because of missing IPv6 support in the native implementation.
```

  ([PR #1065](https://github.com/metallb/metallb/pull/1065))

> - In FRR Mode, it is possible to have a BGP session paired with a BFD session

Reviewer: ```suggestion
- In FRR mode, it is possible to have a BGP session paired with a BFD session
```

  for quicker path failure detection.
  ([PR #927](https://github.com/metallb/metallb/pull/927))
  ([PR #967](https://github.com/metallb/metallb/pull/967))

> - A new manifest, `metallb-frr.yaml` is available to deploy metallb in FRR mode

Reviewer: ```suggestion
- A new manifest (`manifests/metallb-frr.yaml`) is available to deploy MetalLB in FRR mode
```

  ([PR #1014](https://github.com/metallb/metallb/pull/1014))

> - (helm chart) Add support for deploying metallb in frr mode.

Reviewer: ```suggestion
- (helm chart) Add support for deploying MetalLB in FRR mode.
```

>   ([PR #1073](https://github.com/metallb/metallb/pull/1073)

Reviewer 2: missing )

- (helm chart) Allow specification of priorityClassName for speaker and controller.

>   ([PR #1099](https://github.com/metallb/metallb/pull/1099)

Reviewer 2: missing )

Changes in behavior:

- The new FRR mode comes with limitations, compared to the native implementation. The most notable are:

  - It is not allowed to have different peers sharing the same ip address but different ports

>   - It is not allowed to have different routerID or different myAsn for different peers. It is   allowed to override the routerID and / or myAsn, but the value must be the same for all the peers.

Reviewer: ```suggestion
  - It is not allowed to have different routerID or different myAsn for different peers. It is allowed to override the routerID and/or myAsn, but the value must be the same for all the peers.
```

>   - In case the bgp peer is multiple hops away from the nodes, the new ebgp-multihop flag must be set.

Reviewer: ```suggestion
  - In case the BGP peer is multiple hops away from the nodes, the new `ebgp-multihop` flag must be set.
```

> - When switching to FRR mode,the FRR image will required to be downloaded, which may require a longer rollout time than usual.

Reviewer 2: missing space "FRR mode,the FRR image"
---------------------------------- ^

Reviewer: +1. It might be worth also mentioning that a migration path is not currently being tested (in CI or manually, or has someone tried that?)

Reviewer: Lior's comment still stands.

  included tools were not usable.
  ([Issue #618](https://github.com/metallb/metallb/issues/618))

> This release includes contributions from alphabet5, Andrea Panattoni, Brian_P, Carlos Goncalves, Carlos Gonçalves, Federico Paolinelli, Graeme Lawes, HeroCC, Ian Roberts, Lior Noy, Marco Geri, Mark Gray, Mohamed S. Mahmoud, Ori Braunshtein, Pumba98, rata, Russell Bryant, Sabina Aledort, Shivamani Patil and Tyler Auerbeck. Thank you!

Reviewer: Woah! Such a privilege being highlighted twice, although not needed :smile: 
I can go by "Carlos Goncalves", avoids potential special character encoding issues

```suggestion
This release includes contributions from alphabet5, Andrea Panattoni, Brian_P, Carlos Goncalves, Federico Paolinelli, Graeme Lawes, HeroCC, Ian Roberts, Lior Noy, Marco Geri, Mark Gray, Mohamed S. Mahmoud, Ori Braunshtein, Pumba98, rata, Russell Bryant, Sabina Aledort, Shivamani Patil and Tyler Auerbeck. Thank you!
```

Author: Lol that's taken from the commit list.
